yellowbrick/features/pcoords.py

```diff
@@ -169,6 +200,37 @@ def draw(self, X, y, **kwargs):
         Called from the fit method, this method creates the parallel
         coordinates canvas and draws each instance and vertical lines on it.
         """
+        # Convert from dataframe
+        if is_dataframe(X):
+            X = X.as_matrix()
+
+        # Choose a subset of samples
```

Comment:
One of the most difficult things about developing YB is figuring out the semantics between the Scikit-Learn API and the matplotlib API. I totally get what you're saying, and I'm happy to go with it to see what happens when we start deriving subclasses.